association_backend/association/api/tests/mutations/test_manage_user_association_subscription.py

```diff
@@ -0,0 +1,69 @@
+from unittest.mock import patch
+
+from ward import test
+
+from association.api.tests.graphql_client import graphql_client
+from association.domain.exceptions import CustomerNotAvailable
+from association.tests.session import db
+
+
+@test("Customer portal url returned")
+async def _(graphql_client=graphql_client, db=db):
+    query = """
+    mutation {
+        manageUserAssociationSubscription {
+            __typename
+            ... on CustomerPortalResponse {
+                __typename
+                billingPortalUrl
+            }
+        }
+    }
+    """
+    with patch(
+        "association.domain.services.manage_user_association_subscription",
+        return_value="https://stripe.com/stripe_test_customer_portal/cus_test_12345",
+    ) as service_mock:
+        response = await graphql_client.query(query, variables={})
+        print(response.data)
+        assert (
+            response.data["manageUserAssociationSubscription"]["__typename"]
+            == "CustomerPortalResponse"
+        )
+        assert (
+            response.data["manageUserAssociationSubscription"]["billingPortalUrl"]
+            == "https://stripe.com/stripe_test_customer_portal/cus_test_12345"
+        )
+        service_mock.assert_called_once()
+
+
+@test("Customer Not Available")
+async def _(graphql_client=graphql_client, db=db):
+    query = """
+    mutation {
+        manageUserAssociationSubscription {
+            __typename
+            ... on CustomerNotAvailableError {
+                __typename
+                message
+            }
+        }
+    }
+    """
+    with patch(
+        "association.domain.services.manage_user_association_subscription",
+        return_value="https://stripe.com/stripe_test_customer_portal/cus_test_12345",
+    ) as service_mock:
```

Comment:
@patrick91 what do you think about this? I am not a big fan of mocking services

Reply:
yeah, if we were testing the view directly we could have a custom context with mocked things.

but I guess here we can mock/patch something at lower level maybe not the service directly?